design/design_stopping_sims.Rmd

```diff
@@ -0,0 +1,229 @@
+                         "nCohorts must be positive scalar")
+
+                 o$result()
+             })
+#validObject(.StoppingMinCohorts())
+
+
+StoppingMinCohorts <- function(nCohorts, name)
+{
+    .StoppingMinCohorts(nCohorts=safeInteger(nCohorts))
+}
+
+```
+
+
+The two options we have to print the stopping rule name later: 
+```{r}
+
+myStopping1 <- StoppingMinCohorts(nCohorts=3)
+
+myStopping1@name
+
+class(myStopping1)
+```
+
+# B. Unpack combined stopping rules
+
+Now, let's assume we have a combination of stopping rules. I'll go with class() for now so that I don't have to adjust all the stopping rule classes. How would we access all the names? 
+
+```{r}
+myStopping2 <- StoppingTargetProb(target=c(0.2, 0.35),
+                                     prob=0.5)
+myStopping3 <- StoppingMinPatients(nPatients=20)
+
+myStopping_combined <- (myStopping1 & myStopping2) | myStopping3
+
+```
+
+
+This is the current implementation, I have tested it with multiple combinations 
+ 
+* (Stopping_Rule1  | Stopping_Rule2 | Stopping_Rule3  | Stopping_Rule4  | Stopping_Rule5)
+* ((Stopping_Rule1  & Stopping_Rule2) | Stopping_Rule3  | Stopping_Rule4  | Stopping_Rule5)
+* Stopping_Rules_Combined3 <- (Stopping_Rule1  | Stopping_Rule2 | Stopping_Rule3  | (Stopping_Rule4  & Stopping_Rule5))
+* (Stopping_Rule1  | (Stopping_Rule2 & Stopping_Rule3  & Stopping_Rule4)  | Stopping_Rule5)
+* Stopping_Rules_Combined5 <- (Stopping_Rule1  |Stopping_Rule2) & (Stopping_Rule3  | Stopping_Rule4)
+
+and it worked fine for all of them. 
+```{r}
+k = 1
+max = 0
+number = list()
+names = list()
+ind = 1
+stopping_objects <- myStopping_combined@stopList
+
+
+
+for ( i in stopping_objects) {
+
+  t <- try(slot(i, name = "stopList"), silent = T)
+
+    if("try-error" %in% class(t)){   # only one rule
+
+
+    if ( max >= k){
+      number[ind] <- as.character(max+1)
+      #names [ind] <- slot(stopping_objects[[ind]], "name")
+      names [ind] <- class(stopping_objects[[ind]])
+      max <- max + 1
+    } else {
+    number[ind] <- as.character(k)
+    #names [ind] <- slot(stopping_objects[[ind]], "name")
+    names [ind] <- class(stopping_objects[[ind]])
+    }
+
+    k = k+1
+    ind = ind + 1
+  } else { # multiple combined rules 
+
+    max = k + length(i@stopList) - 1
+   
+   number[ind] <- paste(k : max, collapse = " & ")
+   #names[ind] <- paste(sapply(stopping_objects[[ind]]@stopList, slot, "name"), collapse = " and ")
+   names[ind]  <- paste(sapply(stopping_objects[[ind]]@stopList, class), collapse = " and ")
+   ind = ind + 1
+   k = k + length(k:max)
+  }
+
+}
+
+names
+number
+
+```
+
+# C. Count number of hits across all iterations and calculate return measures
+
+So, this is how we can get the order and names of the stopping rules for reporting later on. To count the frequency we have to get the information per iteration from the simulation method: 
+
+```{r, eval = F}
+stopLength <-length(object@stopping@stopList)
```

Comment:
This part I don't understand very well yet. I think if you just want the matrix of simulations in rows and the stopping rules in columns filled with `TRUE` and `FALSE` then we should try to get that / adapt the `stopTrial` method, as that one is already going through all the stopping rules and evaluating them.